packages/taiko-client/preconfapi/builder/calldata.go

```diff
@@ -32,30 +33,41 @@ func NewCalldataTransactionBuilder(
 // return an unsigned transaction, intended for preconfirmations.
 func (b *CalldataTransactionBuilder) BuildUnsigned(
 	_ context.Context,
-	txListBytes []byte,
-	l1StateBlockNumber uint32,
-	timestamp uint64,
-	coinbase common.Address,
-	extraData [32]byte,
+	opts BuildUnsignedOpts,
 ) (*types.Transaction, error) {
-	compressedTxListBytes, err := utils.Compress(txListBytes)
-	if err != nil {
-		return nil, err
-	}
+	encodedParams := make([][]byte, 0)
 
-	// ABI encode the TaikoL1.proposeBlock parameters.
-	encodedParams, err := encoding.EncodeBlockParams(&encoding.BlockParams{
-		ExtraData:          extraData,
-		Coinbase:           coinbase,
-		Signature:          []byte{}, // no longer checked
-		L1StateBlockNumber: l1StateBlockNumber,
-		Timestamp:          timestamp,
-	})
-	if err != nil {
-		return nil, err
+	txLists := make([][]byte, 0)
```

Comment:
Just future improvement maybe, but I always found it a bit strange how this is a completely separate path from the blobs. Could this be merged with the blob path eventually where the code just picks one or the other depending on gas/blob gas cost eventually?

Reply:
yeah definitely, right now we basically just start our proposer in one or the other mode, it can definitely just estimate fee and pick the smarter one later